handshake_server.go

```diff
@@ -356,6 +369,24 @@ Curves:
 		}
 	}
 
+	if hs.clientHello.delegatedCredentials {
```

Comment:
This logic should most likely be moved fully into GetCertificate. A cleaner design would be to simply create a new PrivateKey in the certificate object that corresponds to the delegated credential private key. That way almost all logic on the server side would be unchanged, aside from the addition of a new opaque delegated credentials object in the serverhello.